src/Diagnostics.RuntimeHost/Controllers/Site/SiteControllerBase.cs

```diff
@@ -25,20 +25,20 @@ public SiteControllerBase(IStampService stampService, ISiteService siteService,
         protected async Task<App> GetAppResource(string subscriptionId, string resourceGroup, string appName, DiagnosticSiteData postBody, DateTime startTime, DateTime endTime)
         {
             string requestId = string.Empty;
-            if(this.Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues requestIds))
+            if (this.Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues requestIds))
             {
                 requestId = requestIds.FirstOrDefault() ?? string.Empty;
             }
 
             App app = new App(subscriptionId, resourceGroup, appName)
             {
                 DefaultHostName = postBody.DefaultHostName,
-                Hostnames = postBody.HostNames.Select(p => p.Name),
+                Hostnames = postBody.HostNames != null ? postBody.HostNames.Select(p => p.Name) : new List<string>(),
```

Comment:
We really need to figure out how to return the kusto query that is run, because if for some reason this is null, it could cause a lot of confusion if there is no data coming back.